server/src/main/java/io/spine/server/entity/Transaction.java

```diff
@@ -285,21 +282,30 @@ private void commitChangedState(@NonValidated S newState) {
             commitAttributeChanges();
             EntityRecord newRecord = createRecord();
             afterCommit(entityBeforeTransaction, newRecord);
-        } catch (ValidationException exception) {  /* Could only happen if the state
+        } catch (ValidationException validationException) {  /* Could only happen if the state
                                                       has been injected not using
                                                       the builder setters. */
-            InvalidEntityStateException invalidStateException = of(exception);
-            rollback(invalidStateException);
-
-            throw invalidStateException;
-        } catch (RuntimeException genericException) {
+            InvalidEntityStateException se = toStateException(validationException);
+            rollback(se);
+            throw se;
+        } catch (InvalidEntityStateException stateException) {
+            rollback(stateException);
+            throw stateException;
+        }
+        catch (RuntimeException genericException) {
```

Comment:
Eh?

Reply:
I assume you meant variable names. Since we have now only two branches, both of variables are now `e`.